arch/x86/core/intstub.S

```diff
@@ -464,3 +465,62 @@ SECTION_FUNC(TEXT, _irq_sw_handler)
 	jmp _interrupt_enter
 
 #endif
+
+#if CONFIG_X86_DYNAMIC_IRQ_STUBS > 0
+z_dynamic_irq_stub_common:
+	/* stub number already pushed */
+	push $z_x86_dynamic_irq_handler
+	jmp _interrupt_enter
+
+/* Create all the dynamic IRQ stubs
+ *
```

Comment:
Nice macro work, FWIW.

Reply:
Drive-by comment: what I did instead with x86_64 is to have every IDT entry point to a separate 5-byte CALL instruction, all of which entered the same handler (well, two: with error code and without error code, but one falls through to the other).  Then you can extract the vector in C code with just a subtract and shift of the return address pushed by the call.

Single-byte-per-entry overhead (and one call instruction) vs. this scheme, but works symmetrically across all interrupts and doesn't need special handling for "dynamic" ones.

Reply:
I think it might be beneficial to unify these implementations for x86 in the fullness of time, I have no particular attachment to this one.